test/run-nools-lib.js

```diff
@@ -19,6 +19,20 @@ const runNoolsLib = ({ c, targets, tasks }) => {
       },
       now: () => new Date(TEST_DATE),
       isTimely: function() { return true; },
+      isFormSubmittedInWindow: function(reports, form, start, end, count) {
```

Comment:
I am also having the problem of `Utils` not being accessible inside functions such as `defaultResolvedIf` in `task-defaults.js` when running the unit tests. It runs fine in the app though as `Utils` is [added ](https://github.com/medic/cht-core/blob/0d4c275b088a92edb5b971d4f40aafcde3f93d42/shared-libs/rules-engine/src/rules-emitter.js#L49)to the scope when compiling `nools`.

Reply:
If you still need help with this one, let me know and we can see it together :)

Reply:
My problem has been fixed with help from @kennsippell  but your concern regarding the possible deviation from the original code remains.

Reply:
Some ideas, but test if they are feasible :) 

- If it doesn't matter to test isFormSubmittedInWindow content, then we can make a mock object and stub a function by using sinon, send this object to runNoolsLib as a Utils to use. Sample:
``` 
** Your test file **

utilsMock = {
  isFormSubmittedInWindow: sinon.stub().returns(true) // or false depending on your test
};
....
const { emitted } = runNoolsLib({c, tasks, targets, utilsMock}); // pass it to the context

expect(utilsMock.isFormSubmittedInWindow.callCount).to.equal(1); // Now we can check this :) 

** runNoolsLib file **

const runNoolsLib = ({ c, targets, tasks, utilsMock }) => {
  const context = {
     ......
     Utils: {...utilsMock}  // check if utilsMock is defined then use it, otherwise use the existent code here. So we don't break other tests. But from new dev & forward use the utilsMock.
      ....
  }
```

- if we care about testing isFormSubmittedInWindow content, then it's better to use the original code. If possible import the original Utils. Sample:
```
// runNoolsLib file
const Utils = require(....) // 

const runNoolsLib = ({ c, targets, tasks}) => {
  const context = {
     ......
     Utils: {
          ....
          isFormSubmittedInWindow: Utils.isFormSubmittedInWindow,
          ...
      }
      ....
  }
```

I usually do the first one when the dependency lib, this case Utils.isFormSubmittedInWindow, has already good unit test coverage. It's verified that piece is already working.

Share with me if you find any other alternative to learn as well :)

Reply:
>  Utils: {...utilsMock}
We also use `Utils.addDate` in `defaultResolvedIf`. Shall we add it too to the `utilsMock`?

Reply:
yes, I think so for more accurate testing, you can add as needed in your testing.